### PR TITLE
pr/rx_wifi_coex_pib_config Added Coex rx request modes to pib and public API.

### PR DESCRIPTION
**Pull request by ankuns**
Not sure if target should be `int/wifi`.

### CODE REVIEW

File: src/nrf_802154_types.h [hunk 1]
```diff
@@ -161,6 +161,22 @@ typedef uint8_t nrf_802154_src_addr_match_t;
...
+ *
+ * Possible values:
+ * - @ref NRF_802154_COEX_RX_REQUEST_MODE_DISABLED,
+ * - @ref NRF_802154_COEX_RX_REQUEST_MODE_ENERGY_DETECTION,
+ * - @ref NRF_802154_COEX_RX_REQUEST_MODE_PREAMBLE,
+ * - @ref NRF_802154_COEX_RX_REQUEST_MODE_DESTINED
+ */
+typedef uint8_t nrf_802154_coex_rx_request_mode_t;
+
+#define NRF_802154_COEX_RX_REQUEST_MODE_DISABLED           0x00    //!< Coex does not request to arbiter in receive mode.
```
Comment (czeslawmakarski) [inline src/nrf_802154_types.h]: Why not enum?

Comment (ankuns) [reply src/nrf_802154_types.h]: This way fits into already used convention.

Comment (hubertmis) [reply src/nrf_802154_types.h]: You never know size of enum in C. It's not safe to use enum in public API definitions

File: src/nrf_802154_pib.c [hunk 2]
```diff
@@ -57,6 +57,7 @@ typedef struct
     bool                 auto_ack    : 1;                      ///< Indicating if auto ACK procedure is enabled.
     bool                 pan_coord   : 1;                      ///< Indicating if radio is configured as the PAN coordinator.
     uint8_t              channel     : 5;                      ///< Channel on which the node receives messages.
+    nrf_802154_coex_rx_request_mode_t coex_rx_request_mode;    ///< Coex request mode for received frames.
```
Comment (jciupis) [inline src/nrf_802154_pib.c]: I hate to say this, but shouldn't it all be indented to match the coex field?

File: src/nrf_802154.c [hunk 3]
```diff
@@ -161,6 +161,42 @@ int8_t nrf_802154_tx_power_get(void)
     return nrf_802154_pib_tx_power_get();
 }
 
+static bool is_rx_request_mode_supported(nrf_802154_coex_rx_request_mode_t mode)
```
[the remark this answers is not in the document]
Comment (ankuns) [reply src/nrf_802154.c]: so let's move this into pib

File: src/nrf_802154_pib.c (hunk 2 above)
Comment (hubertmis) [reply src/nrf_802154_pib.c]: I didn't want to say it loud. I hoped Travis would do that instead :)

**Pull request closed without merging**